api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto

```diff
@@ -450,6 +452,34 @@ message HttpConnectionManager {
   // with `prefix` match set to `/dir`. Defaults to `false`. Note that slash merging is not part of
   // `HTTP spec <https://tools.ietf.org/html/rfc3986>` and is provided for convenience.
   bool merge_slashes = 33;
+
+  // Configuration of local reply send by envoy for http responses. Allows to define custom rules
+  // which maps response attribute returned by envoy to different one. Currently mapping of
+  // status code is implemented.
+  message SendLocalReplyConfig {
+    // Configuration which allows to define custom rules for matching local response.
+    // It works like and operator e.g: if you define status code 404 and body pattern
+    // then response must match both
+    message Match {
+      // Definition of status codes to match.
+      repeated uint32 status_codes = 1;
```

Comment:
I would model this as effectively a repeated oneof of individual match types, with the final match being a logical AND of all of the conditions.

Reply:
Changed it